rules/integrations/aws/privilege_escalation_attached_malicious_lambda_layer.toml

```diff
@@ -0,0 +1,57 @@
+[metadata]
+creation_date = "2021/10/03"
+maturity = "production"
+updated_date = "2021/10/03"
+integration = "aws"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies when an user attached a Lambda layer to an existing function to override a library that is in use by the 
+function, where their malicious code could utilize the function's IAM role for AWS API calls. This would give an 
+adversary access to the privileges associated with the Lambda service role that is attached to that function.
+"""
+false_positives = [
+    """
+    Lambda Layer being attached may be performed by a system administrator. Verify whether the user identity, user 
+    agent, and/or hostname should be making changes in your environment. Lambda Layer being attached from unfamiliar
+    users should be investigated. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-25m"
+index = ["filebeat-*", "logs-aws*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS Attached Malicious Lambda Layer"
+note = """## Config
+
+The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://docs.aws.amazon.com/lambda/latest/dg/API_UpdateFunctionConfiguration.html",
+    "https://rhinosecuritylabs.com/aws/aws-privilege-escalation-methods-mitigation-part-2/",
+]
+risk_score = 21
+rule_id = "59ada119-64f4-466f-be04-7a123f5a107a"
+severity = "low"
+tags = ["Elastic", "Cloud", "AWS", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:aws.cloudtrail and event.provider:lambda.amazonaws.com and 
+event.action:UpdateFunctionConfiguration and event.outcome:success
```

Comment:
@w0rk3r 

I will need to spin up a new environment (I create new elastic instances when I'm testing) and recreate.

But I know I forgot something... that an existing lambda function will have a role attaced so `AttachUserPolicy`

I don't know if it would better to just leave the event provider out  `event.provider:(lambda.amazonaws.com or iam.amazonaws.com)` if the rule has two event actions for different providers

**WITHOUT**
```
event.dataset:aws.cloudtrail and event.action:(UpdateFunctionConfiguration and 
AttachUserPolicy) and event.outcome:success
```
or

**WITH**
```
event.dataset:aws.cloudtrail and event.provider:(lambda.amazonaws.com or iam.amazonaws.com) and 
event.action:(UpdateFunctionConfiguration and AttachUserPolicy)and event.outcome:success
```

Reply:
We can work on a sequence using EQL if needed, or maintain like that and add a filter if possible